themes/origins/assets/component-countdown-timer.js

```diff
@@ -0,0 +1,94 @@
+if (!customElements.get("yc-countdown-timer")) {
+  class CountdownTimer extends HTMLElement {
+    static MS_PER_SECOND = 1000;
+    static MS_PER_MINUTE = 60 * CountdownTimer.MS_PER_SECOND;
+    static MS_PER_HOUR = 60 * CountdownTimer.MS_PER_MINUTE;
+    static MS_PER_DAY = 24 * CountdownTimer.MS_PER_HOUR;
+
+    constructor() {
+      super();
+      const [days, hours, minutes, seconds] = this.querySelectorAll(
+        "#days, #hours, #minutes, #seconds",
+      );
+      this.days = days;
+      this.hours = hours;
+      this.minutes = minutes;
+      this.seconds = seconds;
+    }
+
+    disconnectedCallback() {
+      if (this.interval) {
+        clearInterval(this.interval);
+      }
+    }
+
+    connectedCallback() {
+      this._render();
+    }
+
+    _render() {
+      const { date, time } = this.dataset;
+      const { paddedDate, paddedTime } = this._getPaddedDateTimeString(
+        date,
+        time,
+      );
+
+      const endDateTime = new Date(`${paddedDate}T${paddedTime}`);
+
+      if ((!endDateTime) instanceof Date || isNaN(endDateTime)) {
+        this._updateUI(0, 0, 0, 0);
+
+        return;
+      }
+
+      this.interval = setInterval(() => {
+        this._countDownFrom(endDateTime);
+      }, 1000);
+    }
+
+    _getPaddedDateTimeString(date, time) {
```

Comment:
use an underscore (_) prefix for functions called in the `connectedCallback` method

```suggestion
    getPaddedDateTimeString(date, time) {
```